lib/crystalball/map_generator/parser_strategy/processor.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'parser/current'
+
+module Crystalball
+  class MapGenerator
+    class ParserStrategy
+      # Parses the given source files and adds the class and module definitions
+      # to the `consts_defined` array.
+      class Processor
+        attr_accessor :consts_defined, :consts_interacted_with, :current_scope
+
+        def on_send(node, _)
+          const = filtered_children(node).detect { |c| c.type == :const }
+          return unless const
+          add_constant_interacted(qualified_name_from_node(const), nil)
+        end
+
+        def on_casgn(node, scope)
+          namespace, name, = node.children
+          scope_name = namespace ? qualified_name(qualified_name_from_node(namespace), scope) : scope
+          add_constant_defined(name, scope_name)
+        end
+
+        def on_class(node, scope)
+          const, _, body = node.children # TODO: _ is superclass
```

Comment:
why `TODO` ? are you going to handle it in some way?

Reply:
Yes, I'll add it to the map as well

Reply:
write a more detailed description in such case. Like what should be done here why it isn't

Reply:
This one still needs a description